# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLMethodDispatcher.h
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f dom/canvas/WebGLMethodDispatcher.h
--- a/dom/canvas/WebGLMethodDispatcher.h	Wed Jun 10 21:57:19 2020 +0000
+++ b/dom/canvas/WebGLMethodDispatcher.h	Wed Jun 10 22:21:02 2020 +0000
@@ -59,22 +59,19 @@
 DEFINE_ASYNC(HostWebGLContext::DeleteTransformFeedback)
 DEFINE_ASYNC(HostWebGLContext::DeleteVertexArray)
 
-DEFINE_ASYNC(HostWebGLContext::ClearVRFrame)
-// DEFINE_SYNC(HostWebGLContext::GetVRFrame)
-
 DEFINE_ASYNC(HostWebGLContext::Disable)
 DEFINE_ASYNC(HostWebGLContext::Enable)
 DEFINE_ASYNC(HostWebGLContext::GenerateError)
 DEFINE_SYNC(HostWebGLContext::GetCompileResult)
+DEFINE_SYNC(HostWebGLContext::GetFrontBufferSnapshot)
 DEFINE_SYNC(HostWebGLContext::GetFragDataLocation)
-// DEFINE_SYNC(HostWebGLContext::GetLinkResult)
-DEFINE_SYNC(HostWebGLContext::InitializeCanvasRenderer)
+DEFINE_SYNC(HostWebGLContext::GetFrontBuffer)
+DEFINE_SYNC(HostWebGLContext::GetLinkResult)
 DEFINE_SYNC(HostWebGLContext::IsEnabled)
 DEFINE_ASYNC(HostWebGLContext::Resize)
 DEFINE_ASYNC(HostWebGLContext::RequestExtension)
 DEFINE_SYNC(HostWebGLContext::DrawingBufferSize)
 DEFINE_SYNC(HostWebGLContext::OnMemoryPressure)
-DEFINE_SYNC(HostWebGLContext::Present)
 DEFINE_ASYNC(HostWebGLContext::DidRefresh)
 DEFINE_SYNC(HostWebGLContext::GetParameter)
 DEFINE_SYNC(HostWebGLContext::GetString)
@@ -111,6 +108,7 @@
 DEFINE_ASYNC(HostWebGLContext::LinkProgram)
 DEFINE_SYNC(HostWebGLContext::PixelStorei)
 DEFINE_ASYNC(HostWebGLContext::PolygonOffset)
+DEFINE_ASYNC(HostWebGLContext::Present)
 DEFINE_ASYNC(HostWebGLContext::SampleCoverage)
 DEFINE_ASYNC(HostWebGLContext::Scissor)
 DEFINE_ASYNC(HostWebGLContext::ShaderSource)